modules/tinymce/src/core/main/ts/dom/StyleSheetLoaderRegistry.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Document as DomDocument, Node as DomNode, ShadowRoot } from '@ephox/dom-globals';
+import { StyleSheetLoader, StyleSheetLoaderSettings } from 'tinymce/core/api/dom/StyleSheetLoader';
+import { Element, ShadowDom } from '@ephox/sugar';
+import { Option } from '@ephox/katamari';
+
+export interface StyleSheetLoaderRegistry {
+  readonly forElement: (referenceElement: Element<DomNode>, settings: Partial<StyleSheetLoaderSettings>) => StyleSheetLoader;
+}
+
+/**
+ * For testing only - please use StyleSheetLoader.instance in production code.
+ */
+export const create = (): StyleSheetLoaderRegistry => {
```

Comment:
Bit strange to have testing code mixed with production code could this live in a test utils module on where the tests live.

Reply:
Comment is probably misleading. This function is production code - it's only exported for testing purposes.

Reply:
I've revised the comment.